scheduler-interface-spec.md

```diff
@@ -826,6 +826,37 @@ message ForgotAllocation {
 }
 ```
 
+#### Event Cache
+
+The Event Cache is a SchedulerPlugin that exposes events about scheduler objects aiming to help the end user to
+see these events from the shim side. An event is sent to the shim side through the callback in a form of an `EventMessage`.
+
+```protobuf
+message EventRecord {
+   enum Type {
+      REQUEST = 0;
+      APP = 1;
+      NODE = 2;
+      QUEUE = 3;
+   }
+
+   // the type of the object associated with the event
+   Type type = 1;
+   // ID of the object associated with the event
+   string objectID = 2;
+   // the group this object belongs to
+   // it specifies the application ID for allocations, the queue for applications and 
+   // it's the name of the partition for queues and nodes
+   string group = 3;
```

Comment:
The message structure needs to stay simple. There could be hundreds of these messages per second. So I agree with @adamantal's comment: no complex set of parameters should be exposed.

The only comment on the fields I have is the same as with `name` that `group` is ambiguous. I would call it a `groupID` It clearly explains that it is an ID that *could* be used to for grouping messages.

For nodes and queues setting it to the partition does not make sense: it is a repeat of a value which will never change. One shim instance can only belong to one partition. It should never get messages that are not for the partition it registered with.
The partition is set when the shim registers. Passing that same value back to the shim each time is just overhead.

Reply:
I just realized in proto3 everything is `optional` by default. Renamed `group` to `groupID` and deleted the comment block for nodes and queues. For those cases the scheduler will simply not populate those fields - this will get rid of the overhead.